### PR TITLE
Add expressive-annotations-validate@2.7.4 w/ npm auto-update

### PR DESCRIPTION
**Pull request by maruilian11**
**git repo url:** https://github.com/jwaliszko/ExpressiveAnnotations
**Watch 16
  Star 130
 Fork 50**
@Amomo could u help me check this PR for #8276 ? thank you!

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.
# Essential checklist
- [ ] I'm the author of this library
  - [ ] I would like to add link to the page of this library on CDNJS on website / readme
- [x] This lib was not found on cdnjs repo
- [x] No already exist / duplicated issue and PR 
- [x] The lib has notable popularity 
  - [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  - [ ] More than 500 downloads stats per month on npm registry 
- [x] Project has public repository on famous online hosting platform (or been hosted on npm)
# Auto-update checklist
- [x] Has valid tags for each versions (for git auto-update) 
- [x] Auto-update setup
- [x] Auto-update target/source is valid.
- [x] Auto-update filemap is correct.
# Git commit checklist
- [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
- [x] The parent of the commit(s) in the PR is not old than 3 days.
- [x] Pull request is sending from a non-master branch with meaningful name.
- [x] Separate unrelated changes into different commits.
- [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
- [x] Close corresponding issue in commit message
- [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by Amomo** [comment]
Hi @PeterDaveHello I need your review!
The version number of "expressive.annotations.validate.js" don't follow the project's version number after "v2.6.6". 
That makes `v2.8.0 of ExpressiveAnnotations project` contains `v2.6.8 of expressive.annotations.validate.js`.

If it is fine keeping this way, I think this PR is ok~

**Comment by maruilian11** [comment]
ping @PeterDaveHello

**Comment by jwaliszko** [comment]
@maruilian11, @Amomo 

Hi,
if I may suggest something, I'd advice to use the version of the script, instead of the version of the product.

I've been just requested to provide npm (https://github.com/jwaliszko/ExpressiveAnnotations/issues/122) and bower (https://github.com/jwaliszko/ExpressiveAnnotations/issues/124) packages as well, and I've justified my thinking related to versioning there.

As you can see the product has changed from 2.8.0 to 2.9.1 since the time this issue has been opened, while the script version is still the same, mainly 2.6.8.

Also, I suggest minor changes to the json file, related to description and tags, to describe the script more accurately, i.e.
```
{
    "name": "expressive-annotations",
    "version": "2.6.8",
    "description": "Client-side component of ExpressiveAnnotations - annotation-based conditional validation library.",
    "filename": "expressive.annotations.validate.js",
    "repository": {
        "type": "git",
        "url": "git://github.com/jwaliszko/ExpressiveAnnotations.git"
    },
    "keywords": [
        "conditional",
        "validation",
        "mvc",
        "dataannotations"
    ],
    "author": "Jarosław Waliszko <jaroslaw.waliszko@gmail.com>",
    "license": "MIT",
    "autoupdate": {
        "source": "git",
        "target": "git://github.com/jwaliszko/ExpressiveAnnotations.git",
        "basePath": "src",
        "files": [
            "expressive.annotations.validate.+(min.js|js|map)"
        ]
    }
}
```
The last thing - the script itself. The version being released should not contain lines [749-760]( https://github.com/jwaliszko/ExpressiveAnnotations/blob/2ccc94f0abe34387a1f18723e638988c7b8bccfe/src/expressive.annotations.validate.js#L749-L760). Normally, they are being removed by [release script](https://github.com/jwaliszko/ExpressiveAnnotations/blob/master/build/release.sh#L53), prior to being published [on github releases](https://github.com/jwaliszko/ExpressiveAnnotations/releases).

I'm open for discussion and any suggestions.

Thanks,
Regards.

**Comment by PeterDaveHello** [comment]
@jwaliszko thanks for your help and explain, I think we should update the package.json and grab the files from GitHub release page as you suggested, one more thing, we'll like to use "auto-update mechanism" on every libraries, so that we don't need to manually add each version, but we don't support to download tarball from GitHub release page and pick files from it, we do support git repo and npm package auto-update, is it possible to put the release files in the tagged commit in the repo? Thank you!

**Comment by jwaliszko** [comment]
@PeterDaveHello, I'd like to do it, but I don't know how shall I proceed with the tagging.

I suppose that in vast majority of cases, the version of cdnjs/npm/bower package directly corresponds to the version of the released JavaScript library. What's more, this sole, integral, library is the whole product at the same time.

Vast majority of repositories are not cross-language though, and versioning is straightforward. In my case we have mixed C#/JavaScript project, which ultimately consists of 3 components: 2 DLLs and one script. 

Each product release contains these 3 components, separately versioned, e.g.
```
product - v2.9.1
    * DLL 1 - v2.7.0
    * DLL 2 - v2.5.1
    * script.js - v2.6.8
```    
Currently, the release tag (v2.9.1) corresponds to the whole product. From what I understand,  cdnjs/npm/bower are JavaScript package managers, so we need separate tag (currently, v2.6.8), somehow associated with the script.js releases only.

But how to separately denote JavaScript release, within mixed C#/JavaScript git repository? Could you guys help me to understand how shall I proceed with the tagging (treat me as a beginner in that matter)? Can the npm auto update mechanism deal with situations like mine, and choose the "appropriate" tag? 

Also, I'd like to avoid creating new GitHub repository for releasing the script only.

Thanks

**Comment by PeterDaveHello** [comment]
@jwaliszko thanks for following up. Not sure if I answered the exactly what you'll like to know, I know that some projects will have a separate repository for npm/bower release purpose, so that you can just care about the js lib version in the repo, the original repo can just use the version of the whole project, and the separate, release purpose repository uses its own version, but as you said that you'll like to avoid creating new GitHub repository for releasing the script only, I wonder If you'll like to publish to npm repository with its correct version, we can just fetch the files from the npm registry, what do you think?

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by maruilian11** [comment]
ping @jwaliszko thanks!

**Comment by sufuf3** [comment]
@maruilian11 How about that let add the files from GitHub?
And this version is a little old, let's update version to v2.9.5, what do you think?
Thank you.

**Comment by jwaliszko** [comment]
@maruilian11: Hi. Sorry for being not responsive for long time. I'll try to push the npm package this week. Also 2.9.5 version is not correct as it is the version of entire product. The script part follows different versioning, i.e. v2.9.5 of the product contains v2.7.3 of expressive.annotations.validate.js.

**Comment by maruilian11** [comment]
Hi @jwaliszko, thanks for reminding. So the npm package version will be correct? If it will, could you please ping us once you push to the npm package? Thank you so much! It's OK if you forget it, we will check npm package a few days later.

**Comment by jwaliszko** [comment]
@maruilian11 - I'm planning to publish npm package called "expressive-annotations-validate", not "expressive-annotations". The package name, as well as the package version will follow the script name and version, i.e. expressive.annotations.validate.js v2.7.3 (we are talking about product v2.9.5).

My thinking is, since npm, like cdnjs, are for JavaScript components only, applying the version of entire product for the package containing only the single part - client-side script, is misleading. 

This approach would follow the same rule I've done for the nuget package containing single server-side component - ExpressiveAnnotations.dll assembly. Package version equals assembly version there 
https://www.nuget.org/packages/ExpressiveAnnotations.dll/. 

The full product works this way that server-side assemblies "feed" the HTML for the client-side script to work at. Working with the script only, without the C# components, has no sense unless you're constructing HTML manually. That's why the product version should be applied only when 3 components are distributed together. 

Do you think it is reasonable?
Thank you.

**Comment by jwaliszko** [comment]
Hi guys. I've published the script to the npm repository: https://www.npmjs.com/package/expressive-annotations-validate. If I missed something please let me know. Just to inform you, when installing (`npm i expressive-annotations-validate`) I get the following output:
```
PS C:\Projects\GIT\ExpressiveAnnotations\build> npm i expressive-annotations-validate
C:\Projects\GIT\ExpressiveAnnotations
`-- expressive-annotations-validate@2.7.4
  +-- jquery@3.2.1
  +-- jquery-validation@1.17.0
  `-- jquery-validation-unobtrusive@3.2.6

npm WARN Invalid version: "$version$"
npm WARN ExpressiveAnnotations No description
npm WARN ExpressiveAnnotations No repository field.
npm WARN ExpressiveAnnotations No README data
npm WARN ExpressiveAnnotations No license field.
```
I don't know why the warnings appear. The `$version$` field which is in the [package.json hosted on git](https://github.com/jwaliszko/ExpressiveAnnotations/blob/master/package.json#L3) is replaced by the proper version number for each release on my local machine. Then `npm publish` is executed so despite the fact proper package.json (with proper version number) was pushed to the npm, the warning is still here during installation. Next, I do agree there is no README file for the client-script specifically, but the LICENSE is distributed with the package so this warning is surprising as well. The repository field warning is also a mystery for me ([it is added](https://github.com/jwaliszko/ExpressiveAnnotations/blob/master/package.json#L13-L16)). I've asked npm support to help me to understand these warnings. Thanks.

**Comment by maruilian11** [comment]
@jwaliszko Thanks! The file structure looks great! Don't worry about the warning, we will take your package.json on git repo for an example and fulfill it. BTW, is it possible to put the old versions to npm, too? Or we can fetch them from the [original repo](https://github.com/jwaliszko/ExpressiveAnnotations). Thank you!

**Comment by jwaliszko** [comment]
@maruilian11 As requested, I've also published 2 older script versions, i.e. 
* **v2.6.8** (being a part of a few product releases, i.e. [v2.7.3](https://github.com/jwaliszko/ExpressiveAnnotations/releases/tag/v2.7.3), [v2.8.0](https://github.com/jwaliszko/ExpressiveAnnotations/releases/tag/v2.8.0), [v2.9.0](https://github.com/jwaliszko/ExpressiveAnnotations/releases/tag/v2.9.0) and [v2.9.1](https://github.com/jwaliszko/ExpressiveAnnotations/releases/tag/v2.9.1)),
* and **v2.7.0** (being a part of a product [v2.9.2](https://github.com/jwaliszko/ExpressiveAnnotations/releases/tag/v2.9.2)). 

I've omitted few script versions in the middle (i.e. **v2.7.1** /prod [v2.9.3](https://github.com/jwaliszko/ExpressiveAnnotations/releases/tag/v2.9.3), **v2.7.2** /prod [v2.9.4](https://github.com/jwaliszko/ExpressiveAnnotations/releases/tag/v2.9.4) and **v2.7.3** /prod [v2.9.5](https://github.com/jwaliszko/ExpressiveAnnotations/releases/tag/v2.9.5)), since there were a frequent product releases between each other, and the number of downloads is not that significant ([see nuget stats](https://www.nuget.org/packages/ExpressiveAnnotations/)). Also, there appeared to be some issues which were then fixed in the latest, i.e. **v2.7.4** /prod [v2.9.6](https://github.com/jwaliszko/ExpressiveAnnotations/releases/tag/v2.9.6) (as of now), which is actually published to npm as well, so I think we should be fine with that. Thanks.

**Comment by maruilian11** [comment]
@jwaliszko Thank you for such a detailed explanation! I think versions on npm would be enough.
@cdnjs/library-reviewer I have updated the PR. Please help me review it. Thank you!

File: ajax/libs/expressive-annotations-validate/package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "expressive-annotations-validate",
+  "filename": "expressive.annotations.validate.js",
```
Comment (sufuf3) [inline ajax/libs/expressive-annotations-validate/package.json]: Let's use `expressive.annotations.validate.min.js`
Thank you.

**Comment by maruilian11** [comment]
@sufuf3 Thanks. I updated for it.

**Comment by sufuf3** [comment]
@maruilian11 Can we add lib from GitHub repo manually?
npm auto-updater only get v2.7.0 & 2.6.8

And I knew that the author mention here https://github.com/cdnjs/cdnjs/pull/8726#issuecomment-333316682
As he said v2.7.1 /prod v2.9.3, v2.7.2 /prod v2.9.4 and v2.7.3 /prod v2.9.5
So maybe we can add v2.7.1(GitHub is v2.9.3), v2.7.2, v2.7.3[1].
What do you think?

[1] https://github.com/jwaliszko/ExpressiveAnnotations/blob/v2.9.5/src/expressive.annotations.validate.js & https://github.com/jwaliszko/ExpressiveAnnotations/blob/v2.9.5/src/expressive.annotations.validate.min.js 

Thank you.

**Comment by maruilian11** [comment]
@sufuf3 I think we should respect the author's opinions.

**Comment by sufuf3** [comment]
@extend1994 @pvnr0082t please help review this PR, thanks.

**Comment by PeterDaveHello** [comment]
@extend1994 I believe because this PR is older than the feature to modify PR, so the checkbox is not checked, if you didn't see the hint like "push to xxx's yyy branch to update the PR", it means we can't modify it, so you can save the time and compute resource to rebase it, instead, remind the contributor to check the checkbox so that we can do it.

**Comment by PeterDaveHello** [comment]
Looks almost good to me, but I want to be sure, so the latest package on npm is v2.7.4 but it's v2.9.6 on GitHub, and it's not a **dated** case?

**Comment by maruilian11** [comment]
Just checked `Allow edits from maintainers.` checkbox.

**Comment by PeterDaveHello** [comment]
@maruilian11 also rebase?

**Comment by maruilian11** [comment]
@PeterDaveHello expressive-annotations-validate@2.7.4 is for ExpressiveAnnotations@2.9.6 
https://github.com/cdnjs/cdnjs/pull/8726#issuecomment-333316682

**Comment by maruilian11** [comment]
Doing the rebase.

**Comment by maruilian11** [comment]
Done

**Comment by PeterDaveHello** [comment]
Thanks!

**Pull request merged**